---
title: "PROMIS GWAS Summary"
author: M.Vujkovic
output: html_document
---

# Study Design
The study population consists of individuals from 2 cohorts (GWAS1, n = 9.226; GWAS2, n = 8.450) of the the Pakistan Risk of Myocardial Infarction Study (PROMIS). Germline DNA was collected and whole-genome genotyping was performed on the Illumina Infinium Beadchip. Imputation was performed using 1000 Genomes Project data for the reference population.

## Phenotypes
- Myocardial infarction (MI)(dichotomous)
- LDL-C mg/dl (continuous)

## Covariates
- age
- gender
- PCA1-10

## Patient Exclusion
Patients were excluded that were marked in the duplicate lists as supplied by JJ (GWAS1, n = 11; GWAS2, n = 8).

# Analysis

## SNPTEST
Imputed data were analyzed using SNPTEST, where the frequentist method was chosen for association. Filtering on output was based on MAF > .01 and .05, HWE > 10xe06, and INFO > .8

## METAL
Results from association in GWAS1 and GWAS2 were meta-analyzed using METAL.

## Visualization
Manhattan and QQ-plots were generated using R-qqman package.

```{r include = FALSE, echo = FALSE, message = FALSE}

args <- commandArgs(trailingOnly = TRUE)

require("ggplot2")
require("qqman")
require("data.table")

dir = "/project/saleheenlab/snakemake/"

# For MAF > .05
s1    <- read.table(paste(dir, "out/PROMIS/imputed/summary/", args[1], "/", args[1], "_gwas1.snptest.filtered.out", sep = ""), sep = " ",  T, stringsAsFactors = F)
s2    <- read.table(paste(dir, "out/PROMIS/imputed/summary/", args[1], "/", args[1], "_gwas2.snptest.filtered.out", sep = ""), sep = " ",  T, stringsAsFactors = F)
metal <- read.table(paste(dir, "out/PROMIS/imputed/summary/", args[1], "/promis_", args[1], ".metal.1.TBL", sep = ""), sep = "\t", T, stringsAsFactors = F)

# rename position and p-value for qqman manhattan plots
colnames(s1)[5]  <- "BP"
colnames(s1)[23] <- "P"
colnames(s2)[5]  <- "BP"
colnames(s2)[23] <- "P"

# only select SNPs that occur in both studies

colnames(d12)[7] <- "P"

```
```{r echo=TRUE}
# insert table here with regional hits
```

```{r, fig.width = 15, fig.height = 5}
manhattan(s1,
        main = paste("PROMIS 1: ", args[1], " Manhattan plot", sep = ""),
        adj = 0,
        xlab = "",
        ylim = c(0, 16),
        cex = 0.5,
        cex.axis = 1,
        cex.main = 2,
        col = c("steelblue", "steelblue1", "slategray3"),
        suggestiveline = -log10(1e-05),
        genomewideline = -log10(5e-08))
```

```{r, fig.width = 7, fig.height = 7}
qq(s1$P,
        main = paste("PROMIS 1: ", args[1], " Q-Q plot", sep = ""),
        xlim = c(0, 16),
        ylim = c(0, 16),
        pch = 18,
        col = "blue4",
        cex = 0.5,
        las = 1)
```

```{r, fig.width = 15, fig.height = 5}
manhattan(s2,
        main = paste("PROMIS 2: ", args[1], " Manhattan plot", sep = ""),
        adj = 0,
        xlab = "",
        ylim = c(0, 16),
        cex = 0.5,
        cex.axis = 1,
        cex.main = 2,
        col = c("steelblue", "steelblue1", "slategray3"),
        suggestiveline = -log10(1e-05),
        genomewideline = -log10(5e-08))
```
```{r, fig.width = 7, fig.height = 7}
qq(s2$P,
        main = paste("PROMIS 2: ", args[1], " Q-Q plot", sep = ""),
        xlim = c(0, 16),
        ylim = c(0, 16),
        pch = 18,
        col = "blue4",
        cex = 0.5,
        las = 1)
```

```{r, fig.width = 15, fig.height = 5}
manhattan(d12,
        main = paste("PROMIS 1+2: ", args[1], " Meta-Manhattan", sep = ""),
        adj = 0,
        xlab = "",
        ylim = c(0, 16),
        cex = 0.5,
        cex.axis = 1,
        cex.main = 2,
        col = c("steelblue", "steelblue1", "slategray3"),
        suggestiveline = -log10(1e-05),
        genomewideline = -log10(5e-08))
```

```{r, fig.width = 7, fig.height = 7}
qq(d12$P,
        main = paste("PROMIS 1+2: ", args[1], " Meta-Q-Q plot", sep = ""),
        xlim = c(0, 16),
        ylim = c(0, 16),
        pch = 18,
        col = "blue4",
        cex = 0.5,
        las = 1)
```

